### PR TITLE
[rfr] Add identity admin client

### LINKED ISSUE #470: Can not get full tenant list
Hi, gophercloud tenants API only get one tenant, not full tenants list.

Code sample:

```go
 identity := openstack.NewIdentityV2(provider)
topts := tenants.ListOpts{}
tpager := tenants.List(identity, &topts)
p, _ := tpager.AllPages()
fmt.Println("Body: ", p.GetBody())
```

### PR DESCRIPTION
**Pull request by feiskyer**
Add identity admin client, so we can do some admin operations.

This PR fixes #470

### CODE REVIEW

**Comment by jrperritt** [comment]
Is this ready for review? Please use `[wip]` and `[rfr]` tags in the issue title so that others know this is either a work in progress or ready for review.

**Comment by feiskyer** [comment]
@jrperritt Renamed.

File: openstack/client.go [hunk 1]
```diff
@@ -197,6 +197,19 @@ func NewIdentityV3(client *gophercloud.ProviderClient) *gophercloud.ServiceClien
 	}
 }
 
+func NewIdentityAdmin(client *gophercloud.ProviderClient) (*gophercloud.ServiceClient, error) {
```
Comment (jrperritt) [inline openstack/client.go]: Is this for v2 or v3 or are they the same?

File: openstack/client.go [hunk 2]
```diff
@@ -197,6 +197,19 @@ func NewIdentityV3(client *gophercloud.ProviderClient) *gophercloud.ServiceClien
...
+	eo := gophercloud.EndpointOpts{
```
Comment (jrperritt) [inline openstack/client.go]: Can the endpoint change by region? There is a region field in the example object in the docs:
```
            {
                "endpoints": [
                    {
                        "adminURL": "http://23.253.72.207:35357/v2.0",
                        "region": "RegionOne",
                        "internalURL": "http://23.253.72.207:5000/v2.0",
                        "id": "26af053673df4ef3a2340c4239e21ea2",
                        "publicURL": "http://23.253.72.207:5000/v2.0"
                    }
                ],
                "endpoints_links": [],
                "type": "identity",
                "name": "keystone"
            }
```

File: openstack/client.go (hunk 1 above)
Comment (feiskyer) [inline openstack/client.go]: The version is read from keystone's existing endpoints.

File: openstack/client.go (hunk 2 above)
Comment (feiskyer) [inline openstack/client.go]: @jrperritt Yes, it reuses the same `EndpointLocator` procedure as other resources.

File: openstack/client.go (hunk 1 above)
Comment (jrperritt) [inline openstack/client.go]: What if both v2 and v3 endpoints exist?

File: openstack/client.go (hunk 2 above)
Comment (jrperritt) [inline openstack/client.go]: Yes, but my question is: If the endpoint can change by region, and I want the endpoint for a particular region, how would I pass in the `region` to specify which endpoint I wanted? Other (non-identity) services pass a `gophercloud.EndpointOpts` to the function, which contains the region.

**Comment by jrperritt** [comment]
Also, sorry for my ignorance. I'm unfamiliar with this OpenStack feature.

File: openstack/client.go (hunk 1 above)
Comment (feiskyer) [inline openstack/client.go]: @jrperritt This is a good question, I'll check.

**Comment by feiskyer** [comment]
@jrperritt You're right. I split this API into V2 and V3.

**Comment by feiskyer** [comment]
CC @jrperritt

**Comment by feiskyer** [comment]
CC @jrperritt again

**Comment by feiskyer** [comment]
CC @jrperritt Could you pls review this?

**Comment by jrperritt** [comment]
This looks good. +2

**Pull request merged**